torchgeo/datasets/substation.py

```diff
@@ -0,0 +1,241 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""This module handles the Substation segmentation dataset."""
+
+import glob
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from .geo import NonGeoDataset
+from .utils import download_url, extract_archive
+
+
+class SubstationDataset(NonGeoDataset):
+    """Base class for Substation Dataset.
+
+    This dataset is responsible for handling the loading and transformation of
+    substation segmentation datasets. It extends NonGeoDataset, providing methods
+    for dataset verification, downloading, and transformation.
+    Dataset Format:
+    * .npz file for each datapoint
+
+    Dataset Features:
+
+    * 26,522 image-mask pairs stored as numpy files.
+    * Data from 5 revisits for most locations.
+    * Multi-temporal, multi-spectral images (13 channels) paired with masks,
+      with a spatial resolution of 228x228 pixels
+
+    If you use this dataset in your research, please cite the following:
+    * https://doi.org/10.48550/arXiv.2409.17363
+    """
+
+    directory: str = 'Substation'
+    filename_images: str = 'image_stack.tar.gz'
+    filename_masks: str = 'mask.tar.gz'
+    url_for_images: str = 'https://storage.googleapis.com/tz-ml-public/substation-over-10km2-csv-main-444e360fd2b6444b9018d509d0e4f36e/image_stack.tar.gz'
+    url_for_masks: str = 'https://storage.googleapis.com/tz-ml-public/substation-over-10km2-csv-main-444e360fd2b6444b9018d509d0e4f36e/mask.tar.gz'
+
+    def __init__(
+        self,
+        data_dir: str,
+        in_channels: int,
```

Comment:
We usually call this `bands` and allow the user to specify a list of bands, which may not be in order